src/main/scala/sbtprotoc/ProtocPlugin.scala

```diff
@@ -625,7 +636,23 @@ object ProtocPlugin extends AutoPlugin {
         }
       )
 
-      def compileProto(): Set[File] =
+      def compileProto(): Set[File] = {
+        val classLoader: BridgeArtifact => ClassLoader = stampedClassLoadersByArtifact
+          .mapValues(_._2)
+          .applyOrElse(
+            _,
+            { artifact: BridgeArtifact =>
+              // Fallback to the resolver on demand if protocbridge requests an artifact that
+              // was not preloaded because it was not directly referenced from PB.targets generators
+              val classpath = resolver(artifact)
+              log.warn(
+                s"Initializing on-demand cacheloader for unknown artifact $artifact - " +
+                  s"consider upgrading sbt-protoc or setting 'PB.recompile := true' to avoid " +
```

Comment:
I am speculating that a future version of `protocbridge` might resolve artifacts other than the ones known by the current sbt-protoc

Reply:
"Consider upgrading sbt-protoc" - to which version? Since this message comes from sbt-protoc, I am not sure if that's what you meant.

Reply:
> "Consider upgrading sbt-protoc" - to which version?

I know it is very vague, but I was trying to be provide a message as actionable as possible. I initially didn't have a fallback to fail verbosely, but I believe the fallback would work fine for most of the users, so I decided to add it with a WARN.

> Since this message comes from sbt-protoc, I am not sure if that's what you meant.

If that line is reached, it means that the protocbridge JAR in the classpath has a future version of https://github.com/scalapb/protoc-bridge/blob/420593936bd8bf871a43fc4f79793cf3c3e566c9/bridge/src/main/scala/protocbridge/ProtocBridge.scala#L41-L42 requesting other artifacts (maybe referenced by a new type of generator, or through something like https://github.com/scalapb/protoc-bridge/pull/157?). In that case, a later version of sbt-protoc would most likely be available with https://github.com/thesamet/sbt-protoc/blob/d3065b8fd22b0c0734ddfdf3bb92791b46fc94d2/src/main/scala/sbtprotoc/ProtocPlugin.scala#L588 updated accordingly, thus the suggested action.

Maybe we could consider turning this warning into an error during scripted execution somehow, so that any upcoming scripted test exercising such a feature triggering a new type of artifact resolution would fail, until the upfront sbt-protoc resolution that this PR introduces is aligned with the protocbridge logic?

Reply:
I suggest to turn this into a failure.

Reply:
c290ccb